rm-datasource/src/main/java/io/seata/rm/datasource/exec/BaseInsertExecutor.java

```diff
@@ -263,14 +264,17 @@ protected boolean containsColumns() {
         final String sql = sequenceable.getSequenceSql(expr);
         LOGGER.warn("Fail to get auto-generated keys, use '{}' instead. Be cautious, statement could be polluted. Recommend you set the statement to return generated keys.", sql);
 
-        ResultSet genKeys;
-        genKeys = statementProxy.getConnection().createStatement().executeQuery(sql);
-        pkValues = new ArrayList<>();
-        while (genKeys.next()) {
-            Object v = genKeys.getObject(1);
-            pkValues.add(v);
+        Connection conn = statementProxy.getConnection();
+        try (Statement ps = conn.createStatement();
+             ResultSet genKeys = ps.executeQuery(sql)) {
+
+            pkValues = new ArrayList<>();
+            while (genKeys.next()) {
+                Object v = genKeys.getObject(1);
+                pkValues.add(v);
+            }
+            return pkValues;
```

Comment:
Why doesn't `genKeys` need to close?

Reply:
Although GC will close it automatically, but it is better to close it manually.

Reply:
![图片](https://user-images.githubusercontent.com/1527893/119845023-8200d380-bf3b-11eb-8bec-58dbf4ef8c50.png)

Reply:
![图片](https://user-images.githubusercontent.com/1527893/119847516-9e9e0b00-bf3d-11eb-94a5-eefed477192c.png)

Reply:
This may be related to the implementation of JDK. Maybe different versions of JDK won't shut down automatically.
In IBM's JDK, it is better to close manually.